### PR TITLE
Merging tautau branch with lightbylight

### PR DESCRIPTION
**Pull request by matt2275**
I changed applyselections to be able to run for our used by adding an extra option with 3 arguments that needs a flag to tautau and used a text file to give the input and output paths. 
The other edits added a single muon trigger boolean function and added muon information to the event processor.

### CODE REVIEW

File: analysis/applySelections.cpp [hunk 1]
```diff
@@ -7,10 +7,15 @@
 #include "PhysObjectProcessor.hpp"
 #include "ConfigManager.hpp"
 #include "EventDisplay.hpp"
+#include <iostream>
+#include <fstream>
+#include <string>
```
Comment (jniedzie) [inline analysis/applySelections.cpp]: Move all standard includes to Helpers.hpp

File: analysis/applySelections.cpp [hunk 2]
```diff
@@ -140,14 +145,54 @@ bool IsPassingAllLbLCuts(Event &event, bool doHighAco)
...
+
+
 /// Application starting point
 int main(int argc, char* argv[])
 {
-  if(argc != 1 && argc != 9 && argc != 5 && argc != 4){
+  if(argc != 1 && argc != 9 && argc != 5 && argc != 4 && argc != 3){
     cout<<"This app requires 0, 1, 2 or 8 parameters."<<endl;
     cout<<"./getEfficienciesData configPath inputPath outputPathReco outputPathTrigger outputPathHFveto outputPathExclusivity outputPathLbLsignal outputPathQEDsignal"<<endl;
     cout<<"or\n"<<endl;
```
Comment (jniedzie) [inline analysis/applySelections.cpp]: Add description of arguments for the new case.

File: analysis/applySelections.cpp [hunk 3]
```diff
@@ -173,19 +217,40 @@ int main(int argc, char* argv[])
...
+  
   if(argc == 4){
     configPath = argv[1];
     inFilePath = argv[2];
     outFilePaths.push_back(argv[3]);
   }
+
+    if(argc == 3){
+    std::string arg_1 = argv[1];
+    std::string arg_2 = argv[2];
```
Comment (jniedzie) [inline analysis/applySelections.cpp]: Here and in general: no need for `std::`, as we have a global `using namespace std` in Helpers.hpp

Also, please give some meaningful names to variables, e.g. arg_2 -> setupFilePath

File: analysis/applySelections.cpp [hunk 4]
```diff
@@ -173,19 +217,40 @@ int main(int argc, char* argv[])
...
+    {
+        arg_list.push_back(str);
+    }
+    configPath = arg_list[0];
+    inFilePath = arg_list[1];
+    outFilePaths.push_back(arg_list[2]); // single mu
+    outFilePaths.push_back(arg_list[3]); // muon electron
+    outFilePaths.push_back(arg_list[4]); // dimuon 
+    }
+  }
```
Comment (jniedzie) [inline analysis/applySelections.cpp]: Fix indentation (please use the same as through the project, which is 1 indent = 2 spaces). Please use consistent styling for brackets: loops, statements etc. - same line, methods/functions - next line.

File: analysis/runfile.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/afs/cern.ch/user/m/mnickel/private/LightByLight2018/analysis/configs/efficiencies.md
+/eos/cms/store/group/phys_heavyions/rchudasa/lbyl_2018/data_ntuples/HIForward/ntuples_data_new/200105_192027/0000/data_HiForestAOD_1-943.root
+/eos/user/m/mnickel/TauTau/testdata/data_sig_mu_1-943-2.root
+/eos/user/m/mnickel/TauTau/testdata/data_mu_ele_1-943-2.root
+/eos/user/m/mnickel/TauTau/testdata/data_mu_mu_1-943-2.root
```
Comment (jniedzie) [inline analysis/runfile.txt]: Could we move this file to some better location, e.g. to configs? Also please give it some more meaningful name.

File: analysis/src/Event.cpp [hunk 6]
```diff
@@ -186,6 +193,68 @@ PhysObjects Event::GetGoodElectrons(TH1D *cutFlowHist)
   return physObjects.at(kGoodElectron);
 }
 
+PhysObjects Event::GetGoodMuons(TH1D *cutFlowHist)
+{
+  if(physObjectsReady.at(kGoodMuon)) return physObjects.at(kGoodMuon);
+  
+  physObjects.at(kGoodElectron).clear();
```
Comment (jniedzie) [inline analysis/src/Event.cpp]: Uu, please don't clear my good electrons when selecting good muons! ;) Also below, please change to kMuon/kGoodMuon

**Comment by matt2275** [comment]
I think I fixed all the issues you raised. Thanks for the help and the excellent framework!

**Pull request merged**